cirq/protocols/decompose.py

```diff
@@ -44,20 +44,28 @@ def _value_error_describing_bad_operation(op: 'cirq.Operation') -> ValueError:
 class SupportsDecompose(Protocol):
     """An object that can be decomposed into simpler operations.
 
-    Returning `NotImplemented` means "not decomposable". Otherwise an operation,
-    list of operations, or generally anything meeting the `OP_TREE` contract can
-    be returned.
-
-    For example, a Toffoli could be decomposed into Hadamards, CNOTs, and Ts.
-    These operations are simpler because they act on fewer qubits. In general, a
-    decomposition should move closer towards the basic 1-qubit and 2-qubit gates
-    included by default in cirq. It is not necessary to get all the way there,
-    just closer (callers will iteratively decompose if needed).
-
-    DECOMPOSITIONS MUST NEVER BE CYCLIC. If B has A in its decomposition, A
-    should not have B in its decomposition. Decomposition is often performed
-    iteratively, until a fixed point or some desired criteria is met. Cyclic
-    decompositions cause that kind of code to loop forever.
+    All decompositions methods should ultimately terminate on basic 1-qubit and
+    2-qubit gates included by default in cirq. For example, it is permitted for
```

Comment:
Which one and two qubit gates?  Clifford Gates?  When we add exponetiated Paulis, those?  What about gates that end up in contrib (which will happen)?

This is not a forward safe definition.

Reply:
Any of those would be fine. Internally we will have some unique end state, but externally people just need to know "get to the internal types".

You might be thinking in terms of the users of `cirq.decompose`? I guess I could leave some phrasing about not waiting for specific gates but instead waiting for e.g. any unitary 2-qubit gate. We could vary the final gates from release to release to really nail this home for people, or just use TwoQubitMatrixGate as the end point.